### PR TITLE
feat: Stabilize Deno.shutdown()

### LINKED ISSUE #9099: Remove `how` parameter from Deno.shutdown()
There are no protocols that meaningfully support a shutdown mode other than ~`SHUT_RD`~ `SHUT_WR`, so this parameter was never too useful to begin with. Tokio 1.0 has dropped the parameter entirely; Deno should do this too. 

Once this is done, we can also stabilize `Deno.shutdown()`.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #9099

### CODE REVIEW

**Review by ry — DISMISSED** [review]
Why not have closeWrite? I'd rather remove Deno.shutdown...

**Comment by bartlomieju** [comment]
> Why not have closeWrite? I'd rather remove Deno.shutdown...

As per @piscisaureus in #9099:
> There are no protocols that meaningfully support a shutdown mode other than SHUT_RD, so this parameter was never too useful to begin with. Tokio 1.0 has dropped the parameter entirely; Deno should do this too.

Tokio has no way of shutting down only the write half.

File: cli/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -1809,6 +1802,18 @@ declare namespace Deno {
...
   export function connectTls(options: ConnectTlsOptions): Promise<Conn>;
 
+  /** Shutdown socket send and receive operations.
+   *
+   * Matches behavior of POSIX shutdown(3).
+   *
+   * ```ts
+   * const listener = Deno.listen({ port: 80 });
+   * const conn = await listener.accept();
+   * Deno.shutdown(conn.rid, Deno.ShutdownMode.Write);
```
Comment (satyarohith) [inline cli/dts/lib.deno.ns.d.ts:1812]:
```suggestion
   * Deno.shutdown(conn.rid);
```

**Comment by bartlomieju** [comment]
> > Why not have closeWrite? I'd rather remove Deno.shutdown...
> 
> As per @piscisaureus in #9099:
> 
> > There are no protocols that meaningfully support a shutdown mode other than SHUT_RD, so this parameter was never too useful to begin with. Tokio 1.0 has dropped the parameter entirely; Deno should do this too.
> 
> Tokio has no way of shutting down only the write half.

Actually now I'm confused reading https://docs.rs/tokio/1.0.1/tokio/io/trait.AsyncWriteExt.html#method.shutdown

> For example, the TcpStream implementation will issue a shutdown(Write) sys call.

**Comment by piscisaureus** [comment]
My mistake. The only mode supported by TCP and Tokio is SHUT_WR.

**Review by piscisaureus — APPROVED** [review]

**Pull request merged**